internal/error.go

```diff
@@ -223,24 +223,26 @@ type (
 	}
 )
 
-// ErrNoData is returned when trying to extract strong typed data while there is no data available.
-var ErrNoData = errors.New("no data available")
+var (
+	// ErrNoData is returned when trying to extract strong typed data while there is no data available.
+	ErrNoData = errors.New("no data available")
 
-// ErrTooManyArg is returned when trying to extract strong typed data with more arguments than available data.
-var ErrTooManyArg = errors.New("too many arguments")
+	// ErrTooManyArg is returned when trying to extract strong typed data with more arguments than available data.
+	ErrTooManyArg = errors.New("too many arguments")
 
-// ErrActivityResultPending is returned from activity's implementation to indicate the activity is not completed when
-// activity method returns. Activity needs to be completed by Client.CompleteActivity() separately. For example, if an
-// activity require human interaction (like approve an expense report), the activity could return activity.ErrResultPending
-// which indicate the activity is not done yet. Then, when the waited human action happened, it needs to trigger something
-// that could report the activity completed event to temporal server via Client.CompleteActivity() API.
-var ErrActivityResultPending = errors.New("not error: do not autocomplete, using Client.CompleteActivity() to complete")
+	// ErrActivityResultPending is returned from activity's implementation to indicate the activity is not completed when
+	// activity method returns. Activity needs to be completed by Client.CompleteActivity() separately. For example, if an
+	// activity require human interaction (like approve an expense report), the activity could return activity.ErrResultPending
+	// which indicate the activity is not done yet. Then, when the waited human action happened, it needs to trigger something
+	// that could report the activity completed event to temporal server via Client.CompleteActivity() API.
+	ErrActivityResultPending = errors.New("not error: do not autocomplete, using Client.CompleteActivity() to complete")
+)
 
-// NewApplicationError create new instance of *ApplicationError with message and optional details.
-func NewApplicationError(message string, nonRetryable bool, cause error, details ...interface{}) *ApplicationError {
+// NewApplicationError create new instance of *ApplicationError with message, type, and optional details.
```

Comment:
Add the explanation that this type is used by retry options.

Reply:
Added this to public alias.